tests/test_schema.py

```diff
@@ -1,278 +1,63 @@
-        model.close()
-
-    with warnings.catch_warnings():
-        warnings.simplefilter('ignore')
-        os.environ['PASS_INVALID_VALUES'] = '0'
-        os.environ['STRICT_VALIDATION'] = '0'
-        model = util.open(TMP_FITS)
-        assert model.meta.instrument.name is None
-        model.close()
-
-    with warnings.catch_warnings():
-        warnings.simplefilter('ignore')
-        os.environ['PASS_INVALID_VALUES'] = '1'
-        model = util.open(TMP_FITS)
-        assert model.meta.instrument.name == 'FOO'
-        model.close()
-
-    del os.environ['PASS_INVALID_VALUES']
-    del os.environ['STRICT_VALIDATION']
-
-    with pytest.raises(validate.ValidationWarning):
-        with warnings.catch_warnings():
-            warnings.simplefilter('error')
-            model = util.open(TMP_FITS,
-                              pass_invalid_values=False,
-                              strict_validation=False)
-            model.close()
-
-    with pytest.raises(jsonschema.ValidationError):
-        model = util.open(TMP_FITS,
-                          pass_invalid_values=False,
-                          strict_validation=True)
-        model.close()
-
-    with warnings.catch_warnings():
-        warnings.simplefilter('ignore')
-        model = util.open(TMP_FITS,
-                          pass_invalid_values=False,
-                          strict_validation=False)
-        assert model.meta.instrument.name is None
-        model.close()
-
-    with warnings.catch_warnings():
-        warnings.simplefilter('ignore')
-        model = util.open(TMP_FITS,
-                          pass_invalid_values=True,
-                          strict_validation=False)
-        assert model.meta.instrument.name == 'FOO'
-        model.close()
-
-    if os.path.exists(TMP_FITS):
-        os.remove(TMP_FITS)
-
-def test_ad_hoc_json():
-    with DataModel() as dm:
-        dm.meta.foo = {'a': 42, 'b': ['a', 'b', 'c']}
-
-        dm.save(TMP_ASDF)
-
-    with DataModel(TMP_ASDF) as dm2:
-        assert dm2.meta.foo == {'a': 42, 'b': ['a', 'b', 'c']}
-
-
-def test_ad_hoc_fits():
+@pytest.mark.parametrize("filename", ["test.asdf", "test.fits"])
+def test_ad_hoc_attributes(filename, tmp_path):
+    """
+    Test that attributes unrecognized by the schema
+    can still be assigned and written.
+    """
+    file_path = tmp_path/filename
     with DataModel() as dm:
         dm.meta.foo = {'a': 42, 'b': ['a', 'b', 'c']}
 
-        dm.to_fits(TMP_FITS, overwrite=True)
+        dm.save(file_path)
 
-    with DataModel.from_fits(TMP_FITS) as dm2:
+    with DataModel(file_path) as dm2:
         assert dm2.meta.foo == {'a': 42, 'b': ['a', 'b', 'c']}
 
 
-@pytest.mark.skip("requires the jwst core schema")
 def test_find_fits_keyword():
-    with DataModel() as x:
-        assert x.find_fits_keyword('DATE-OBS') == \
-          ['meta.observation.date']
+    with FitsModel() as x:
+        assert x.find_fits_keyword('TELESCOP') == ['meta.telescope']
```

Comment:
I know the code for `find_fits_keyword` is in `schema.py`, but it might worth moving the code and the test for it into `fits_support.py` and `test_fits_support.py` at some point.

Reply:
https://github.com/spacetelescope/stdatamodels/issues/33